docs/articles/documentation/test-api/authentication/user-roles.md

```diff
@@ -0,0 +1,118 @@
+
+## Creating and Using Roles
+
+Use the `Role` constructor to create and initialize a role.
+
+```text
+Role( func( t ) )
+```
+
+Parameter | Type     | Description
+--------- | -------- | --------------------------------------------------------------------------------
+`func`    | Function | An asynchronous function that contains logic that authenticates the user.
+`t`       | Object   | The [test controller](../test-code-structure.md#test-controller) used to access test run API.
+
+```js
+import { Role } from 'testcafe';
+
+const regularAccUser = Role(async t => {
+    await t
+        .typeText('#login', 'TestUser')
+        .typeText('#password', 'testpass')
+        .click('#sign-in');
+});
+
+const facebookAccUser = Role(async t => {
+    await t
+        .click('#sign-in-with-facebook')
+        .typeText('#email', 'testuser@mycompany.com')
+        .typeText('#pass', 'testpass')
+        .click('#submit');
+});
+
+const admin = Role(async t => {
+    await t
+        .typeText('#login', 'Admin')
+        .typeText('#password', 'adminpass')
+        .click('#sign-in');
+});
+```
+
+> Role initialization code is executed only once when the role is used for the first time.
+
+After you create the roles, you can switch between users at any moment except for the role initialization code.
+
+To switch to a role, use the `t.useRole` function.
+
+```text
+t.useRole( role )
+```
+
+Parameter | Type   | Description
+--------- | ------ | ---------------------------------------------
+`role`    | Object | The role you need to use further in the test.
+
+```js
+import { Selector } from 'testcafe';
+
+const entry        = Selector('#entry');
+const removeButton = Selector('#remove-entry');
+
+fixture `My Fixture`
+    .page `http://example.com`;
+
+test('test that involves two users', async t => {
+    await t
+        .useRole(regularAccUser)
+        .expect(entry.exists).ok()
+        .expect(removeButton.visible).notOk()
+        .useRole(admin)
+        .expect(removeButton.visible).ok()
+        .click(removeButton)
+        .expect(entry.exists).notOk()
+});
+```
+
+Roles can be shared across tests and fixtures. You can create roles in a separate helper file and use them in any test fixture that references this file.
+
+You can think of a role as a set of cookies. The role saves cookies that were set by its initialization code. When you switch to the role,
+cookies stored in it are set for the current document, so that you become an authenticated user.
+
+If you switch to a role and then authenticate to a different website in test code, cookies set by such authentication are also stored
+in the role. The next time you switch to this role, you will be authenticated to both sites. Note that this is true only within a single test run. If you use this role from a different test, you will be authenticated to just one website.
```

Comment:
I don't like phrasing here, especially `you will be authenticated to just one website`. That's not true if role was initialized for more than one site.